src/main/java/com/cloudbees/jenkins/support/configfiles/SecretHandler.java

```diff
@@ -0,0 +1,90 @@
+package com.cloudbees.jenkins.support.configfiles;
+
+import hudson.util.Secret;
+import org.apache.commons.io.FileUtils;
+import org.xml.sax.Attributes;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+import org.xml.sax.XMLReader;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.transform.OutputKeys;
+import javax.xml.transform.Result;
+import javax.xml.transform.Source;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerException;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.sax.SAXSource;
+import javax.xml.transform.stream.StreamResult;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.StringReader;
+
+import org.xml.sax.helpers.XMLFilterImpl;
+import org.xml.sax.helpers.XMLReaderFactory;
+
+/**
+ * Secret Handler for xml files to add to the support bundle.
+ * We want to use a place holder instead of Secrets.
+ */
+public class SecretHandler {
+
+    /**
+     * our placeholder
+     */
+    protected static final String SECRET_MARKER = "#secret#";
+
+    /**
+     * fine the secret in the xml file and replace it with the place holder
+     * @param xmlFile we want to parse
+     * @return the patched xml files without secrets
+     * @throws ParserConfigurationException
+     * @throws SAXException
+     * @throws IOException
+     * @throws XMLStreamException
+     * @throws TransformerException
+     */
+    public static File findSecrets(File xmlFile) throws ParserConfigurationException, SAXException, IOException, XMLStreamException, TransformerException {
```

Comment:
I wonder if we shouldn't better hide those exceptions here or at least handle some locally, to rule out the comment I made above about using a catch(Exception) when calling that method.